lib/ecto/query/planner.ex

```diff
@@ -215,6 +215,7 @@ defmodule Ecto.Query.Planner do
     query
     |> prepare_sources(adapter)
     |> prepare_assocs
+    |> rewrite_aliases
```

Comment:
@michalmuskala - I am new to elixir and am not familiar with internal workings of ecto. I am not too sure if / how we could do this in builder. I, however, think that it would be better to traverse only if the schema uses field aliases and needs rewrite. I think this should be doable and would work on it.